src/contract/polls.cpp

```diff
@@ -0,0 +1,618 @@
+#include <boost/algorithm/string/case_conv.hpp> // for to_lower()
+#include <boost/algorithm/string.hpp>
+
+#include "main.h"
+#include "polls.h"
+#include "contract.h"
+#include "bitcoinrpc.h"
+#include "appcache.h"
+#include "cpid.h"
+#include "init.h" // for pwalletMain
+#include "block.h"
+
+double GetTotalBalance();
+std::string TimestampToHRDate(double dtm);
+double CoinToDouble(double surrogate);
+double DoubleFromAmount(int64_t amount);
+std::string PubKeyToAddress(const CScript& scriptPubKey);
+bool GetEarliestStakeTime(std::string grcaddress, std::string cpid);
+CBlockIndex* GetHistoricalMagnitude(std::string cpid);
+StructCPID GetLifetimeCPID(const std::string& cpid, const std::string& sFrom);
+bool WalletOutOfSync();
+
+std::pair<std::string, std::string> CreatePollContract(std::string sTitle, int days, std::string sQuestion, std::string sAnswers, int iSharetype, std::string sURL)
+{
+    if (pwalletMain->IsLocked())
+        return std::make_pair("Error", "Please fully unlock the wallet first.");
+    else if (fWalletUnlockStakingOnly)
+        return std::make_pair("Error", "Wallet is unlocked for staking only, needs to be fully unlocked for voting");
+    else if (WalletOutOfSync())
+        return std::make_pair("Error", "Wallet is not synced, you need to be synced in order to create a poll");
```

Comment:
These returnvalue error indicators should be replaced later by exceptions. Not needed fot this PR.